### PR TITLE
Use k8s 1.29 client libs

### PR DESCRIPTION
**Pull request by tamalsaha**

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -8,9 +8,9 @@ import (
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/client-go/rest"
 	clusterv1client "open-cluster-management.io/api/client/cluster/clientset/versioned"
-	clusterv1beta2 "open-cluster-management.io/api/cluster/v1beta2"
 	"open-cluster-management.io/cluster-proxy/pkg/generated/clientset/versioned"
 	"open-cluster-management.io/cluster-proxy/pkg/util"
+	clustersdk "open-cluster-management.io/sdk-go/pkg/apis/cluster/v1beta2"
```
Comment (xuezhaojun) [inline client/client.go:13]: @qiujian16 
Looking for advise for this one, we seems having the 'clusterv1beta2' name style for importing a pkg for a long time and across repos.  When using a api lib, do we need to specify the version?

File: go.mod [hunk 2]
```diff
@@ -1,25 +1,28 @@
 module open-cluster-management.io/cluster-proxy
 
-go 1.20
+go 1.21
+
+toolchain go1.22.0
```
Comment (xuezhaojun) [inline go.mod:5]: How about seperate an independent PR for just upgrade go version to 1.22?

File: client/client.go (hunk 1 above)
Comment (qiujian16) [reply client/client.go:13]: yes, I think so

Comment (tamalsaha) [reply client/client.go]: Updated

File: go.mod (hunk 2 above)
Comment (tamalsaha) [reply go.mod:5]: https://github.com/open-cluster-management-io/cluster-proxy/pull/176

Once merged, I can rebase this pr

File: pkg/proxyagent/agent/agent.go [hunk 3]
```diff
@@ -31,6 +31,7 @@ import (
 	"open-cluster-management.io/cluster-proxy/pkg/config"
 	"open-cluster-management.io/cluster-proxy/pkg/proxyserver/operator/authentication/selfsigned"
 	"open-cluster-management.io/cluster-proxy/pkg/util"
+	clusterv1beta2sdk "open-cluster-management.io/sdk-go/pkg/apis/cluster/v1beta2"
```
Comment (xuezhaojun) [inline pkg/proxyagent/agent/agent.go:34]: Can we align the naming style with this line: https://github.com/open-cluster-management-io/clusteradm/blob/393f050c99ad3614567d4ea5650377876dcf8543/pkg/cmd/get/clusterset/exec.go#L20

File: cmd/Dockerfile [hunk 4]
```diff
@@ -1,21 +1,20 @@
...
-RUN GOOS=linux GOARCH=amd64 CGO_ENABLED=0 \
-      wget https://github.com/kubernetes-sigs/apiserver-network-proxy/archive/refs/tags/v${APISERVER_NETWORK_PROXY_VERSION}.tar.gz \
+RUN wget https://github.com/kubernetes-sigs/apiserver-network-proxy/archive/refs/tags/v${APISERVER_NETWORK_PROXY_VERSION}.tar.gz \
       && tar xzvf v${APISERVER_NETWORK_PROXY_VERSION}.tar.gz \
       && cd apiserver-network-proxy-${APISERVER_NETWORK_PROXY_VERSION} \
-      && go build -o /workspace/proxy-server ./cmd/server/ \
-      && go build -o /workspace/proxy-agent ./cmd/agent/ \
-      && go build -o /workspace/proxy-test-client ./cmd/test-client/ \
-      && go build -o /workspace/proxy-test-server ./cmd/test-server/ \
+      && CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o /workspace/proxy-server ./cmd/server/ \
```
[the remark this answers is not in the document]
Comment (xuezhaojun) [reply cmd/Dockerfile:14]: ACK.

**Comment by xuezhaojun** [comment]
/approve
The changes are in general good, the only concern is whether we can upgrade to 1.21 instead of 1.22.

**Comment by xuezhaojun** [comment]
Hi, @tamalsaha  I noticed the PR seems mean to make cluster-proxy to work with the multicluster-controlplane.

Consulted with @skeeey, in theory it would work, but we didn't have test cases cover this case. The lib problem may not be the only issue you will encounter with.

To better trace and support this case, I created an issue under the multicluster-controlplane repo, please feel free to share any thoughts about this case. 

And we are also looking forward to have you sharing in our bi-week community meeting as well!

File: pkg/proxyagent/agent/agent.go (hunk 3 above)
Comment (tamalsaha) [reply pkg/proxyagent/agent/agent.go]: Done

**Comment by tamalsaha** [comment]
Switched to Go 1.21

**Pull request merged**